### DISCUSSION TITLE
Getting ESP_ERR_INVALID_STATE when initializing and using DAC

### DISCUSSION THREAD

**Discussion by pn2new** [micropython/micropython#16372] [ESP32]
In a micropython program running on an ESP32, I regularly get `OSError: (-259, 'ESP_ERR_INVALID_STATE')` when using the DAC.

This does not occur when I run the program for the first time (after a power-up reset), but occurs reproducibly when I re-run it during program development (it is started using pyboard / repl).

A simple reduced test case looks like this:

```
DAC_PIN = const(25)
dacpin = machine.Pin(DAC_PIN, machine.Pin.OUT)
dac = machine.DAC(dacpin)

while True:
    for i in range(256):
        print("DAC={}".format(i))
        dac.write(int(i))
        time.sleep_ms(1000)
```

What do I need to do to properly initialize the ESP32 when re-starting the program?

**Comment by davefes** [comment]
Have you read this one?
https://github.com/orgs/micropython/discussions/16291

There is a `uart.deinit()` which I have used.  As both the DAC and UART on the ESP32 chip you would think there would be a `dac.deinit()` ... if it is required.

Until someone sorts that out maybe my `poweron_reset` circuit in the link gets you going again.i
